mozdownload/scraper.py

```diff
@@ -892,7 +891,13 @@ def get_build_info_for_index(self, build_index=None):
 
         # If no index has been given, set it to the last build of the day.
         if build_index is None:
-            build_index = len(parser.entries) - 1
+            # Find the most recent non-empty entry.
+            build_index = len(parser.entries)
+            for build in reversed(parser.entries):
+                build_index -= 1
+                if not build_index or self.is_build_dir(build):
+                    break
+
```

Comment:
This looks fine to me but I miss a NotFoundError thrown as in line 888 in case of no build has been found. Right now we would return an invalid build index. I think we can add a test for that simply via `mozdownload -t tinderbox -p linux64 --branch mozilla-inbound --ext rr` because there won't exist such a file.